tools/libeosshellcontent.py

```diff
@@ -0,0 +1,199 @@
+#!/usr/bin/python3
+#
+# Copyright (C) 2017 Endless Mobile, Inc.
+#
+# Authors:
+#  Joaquim Rocha <jrocha@endlessm.com>
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, write to the Free Software Foundation, Inc.,
+# 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+
+import gi
+gi.require_version('AppStreamGlib', '1.0')
+from gi.repository import AppStreamGlib
+from gi.repository import Gio
+import json
+import os
+import polib
+import sys
+import xml.dom.minidom
+import xml.etree.ElementTree as ET
+
+CMS_GS_BUCKET_URL = 'https://d3lapyynmdp1i9.cloudfront.net'
+DEFAULT_HOMEPAGE = 'https://endlessm.com'
+
+def load_as_app_from_appdata(appdata_file):
+    app = AppStreamGlib.App()
+    app.parse_file(appdata_file, AppStreamGlib.AppParseFlags.NONE)
+    return app
+
+class ShellContent:
+
+    def __init__(self, content_dir):
+        self._content_dir = content_dir
+        self._translations, self._langs = self._get_translations_dict()
+
+    def _get_translations_dict(self):
+        """Get translations dictionary and list of languages.
+
+        It reads the translations from a predefined file and returns a dictionary
+        with the translated strings and a list of the available languages.
+        """
+        po_dir = os.path.join(self._content_dir, 'po')
+        strings_dict = {}
+        langs = []
+
+        with open(os.path.join(po_dir, 'LINGUAS')) as linguas:
+            langs = linguas.read().splitlines()
+
+        for lang in langs:
+            po = polib.pofile(os.path.join(po_dir, '{}.po'.format(lang)))
+            strings_dict[lang] = {(entry.msgid, entry.msgctxt): entry.msgstr for entry in po}
+
+        return (strings_dict, langs)
+
+    def _get_app_metadata(self, app_id):
+        metadata_file = os.path.join(self._content_dir, 'content', 'Default',
+                                     'apps', 'content.json')
+        try:
+            all_metadata = json.load(open(metadata_file))
+        except FileNotFoundError:
+            return None
```

Comment:
Why is this exception being swallowed? I'd think we'd want the process to fail if content.json could not be found.